contracts/EasyAuction.sol

```diff
@@ -335,6 +337,84 @@ contract EasyAuction is Ownable {
         public
         atStageSolutionSubmission(auctionId)
     {
+        bytes32 initialAuctionOrder =
+            auctionData[auctionId].initialAuctionOrder;
+        (, , uint96 fullAuctionedAmount) = initialAuctionOrder.decodeOrder();
+
+        uint256 currentBidSum = auctionData[auctionId].interimSumBidAmount;
+        uint256 previousBidSum;
+        bytes32 currentOrder = auctionData[auctionId].interimOrder;
+        bytes32 previousOrder;
+        // same loop as function before
+        uint256 buyAmountOfIter;
+        uint256 sellAmountOfIter;
+        // note: all order have a price that is better than the auction min price
+        do {
+            console.log("loop");
+            previousOrder = currentOrder;
+            previousBidSum = currentBidSum;
+            currentOrder = sellOrders[auctionId].next(currentOrder);
+            (, buyAmountOfIter, sellAmountOfIter) = currentOrder.decodeOrder();
+            currentBidSum = currentBidSum.add(sellAmountOfIter);
+        } while (
+            (currentOrder != IterableOrderedOrderSet.QUEUE_END) &&
+                (currentBidSum.mul(buyAmountOfIter) <
+                    fullAuctionedAmount.mul(sellAmountOfIter))
+        );
+
+        if (currentOrder == IterableOrderedOrderSet.QUEUE_END) {
+            console.log("end of orders reached");
+            // end of order loop reached, partial match
+            (, uint96 auctioneerBuyAmount, ) =
+                initialAuctionOrder.decodeOrder();
+
+            // This order matches auctionData[auctionId].initialAuctionOrder
+            // exactly (note that numerator and denominator are inverted).
+            // It enforces the price to be the same as the initial auction
+            // order, since no better price could be determined.
+            auctionData[auctionId].clearingPriceOrder = IterableOrderedOrderSet
+                .encodeOrder(
+                0xffffffffffffffff, // max user ID. user id is not relevant
+                fullAuctionedAmount,
+                auctioneerBuyAmount
+            );
+
+            auctionData[auctionId].volumeClearingPriceOrder = (
+                previousBidSum.mul(fullAuctionedAmount).div(auctioneerBuyAmount)
+            )
+                .toUint96();
+        } else {
+            console.log("terminated with spare orders");
+            // more orders available than tokens auctiones
+            // cuorrent order may be partially matched
+            // by exit condition must be
+            // currentBidSum * buyAmountOfIter >= fullAuctionedAmount * sellAmountOfIter
+            // previousBidSum * buyAmountOfIter < fullAuctionedAmount * sellAmountOfIter
+            // there exists x s.t. 0 < x <= sellAmountOfIter and
+            // (previousBidSum + x) * buyAmountOfIter = fullAuctionedAmount * sellAmountOfIter
+            auctionData[auctionId].clearingPriceOrder = currentOrder;
```

Comment:
Clearing price could also be a price not directly lying on an order. how do you plan to deal with that situation? 

Sure, this is possible here.... just needs to be done somehow.